dlt/destinations/impl/filesystem/filesystem.py

```diff
@@ -123,16 +124,51 @@ def write(self) -> None:
         from dlt.common.libs.deltalake import (
             write_delta_table,
             _deltalake_storage_options,
+            try_get_deltatable,
         )
 
         file_paths = [job.file_path for job in self.table_jobs]
 
-        write_delta_table(
-            path=self.client.make_remote_uri(self.make_remote_path()),
-            data=pa.dataset.dataset(file_paths),
-            write_disposition=self.table["write_disposition"],
-            storage_options=_deltalake_storage_options(self.client.config),
-        )
+        if (
+            self.table["write_disposition"] == "merge"
+            and (
+                dt := try_get_deltatable(
+                    self.client.make_remote_uri(self.make_remote_path()),
+                    storage_options=_deltalake_storage_options(self.client.config),
+                )
+            )
+            is not None
+        ):
+            assert self.table["x-merge-strategy"] in self.client.capabilities.supported_merge_strategies  # type: ignore[typeddict-item]
```

Comment:
is this just for dev purposes or do you want to leave this here? we might want to extend the mechanism in the loader to check for non-loadable jobs (similar to the way we check for incorrect fileformats); but maybe not for now.

Reply:
Ultimately this should be removed.

This check happens in `_verify_destination_capabilities` for the other destinations, but `filesystem` is exempted there because it has [this](https://github.com/dlt-hub/dlt/pull/1600#discussion_r1680069255) (IMO undesired) behavior.